### ISSUE TITLE
Command line completion

### ISSUE THREAD

**Issue by procrypt** [in-toto/in-toto#115]
It would be nice to have bash completion support for `in-toto` command line interface. I was trying out the [demo application](https://github.com/in-toto/in-toto/tree/develop/demo) and found it that the bash completion is not present.
Adding bash completion would be a good feature IMO and I'll be happy to take it as a task.
Thoughts @lukpueh @JustinCappos @SantiagoTorres

**Comment by SantiagoTorres** [comment]
Doesn't sound bad in my book. But we'd have to ship it somehow, and for now we only have a python module. Any ideas on how to move forward with it?

**Comment by procrypt** [comment]
@SantiagoTorres we can ship a bash file with the the whole package and it'll do the work.

**Comment by vladimir-v-diaz** [comment]
I remember seeing a tab completer (written in Python) for the Seattle project.  You might want to take a look...

https://github.com/SeattleTestbed/seash/blob/3f47692aee94d7c7c8a615b662a23ae7c86bd7d8/tab_completer.py

**Comment by vladimir-v-diaz** [comment]
Oh, if you need tab completion for the argument options (I was thinking in-toto had a custom command-line interface), there should be libraries for that.  Here is one:
http://argcomplete.readthedocs.io/en/latest/

**Comment by SantiagoTorres** [comment]
> @SantiagoTorres we can ship a bash file with the the whole package and it'll do the work.

The file should be placed under /usr/share/bash-completions or so, depending on the distribution/OS. That's what I was worried about

**Comment by SantiagoTorres** [comment]
> I remember seeing a tab completer (written in Python) for the Seattle project. You might want to take a look...

> https://github.com/SeattleTestbed/seash/blob/3f47692aee94d7c7c8a615b662a23ae7c86bd7d8/tab_completer.py

We already have something like this in  #4. I think it predates the seattle autocompleter.

I think that https://argcomplete.readthedocs.io/en/latest/ is rather hacky, as it does oh-my-zsh-like completions under bash...

**Comment by lukpueh** [comment]
FYI, this is also what [git does](https://git-scm.com/book/en/v1/Git-Basics-Tips-and-Tricks#Auto-Completion), i.e. *"Download X from Y and place it under Z"*.
In general, I think bash-completion for in-toto's command line tools is a nice-to-have side-project but there might be more pressing things to do.

**Comment by SantiagoTorres** [comment]
> FYI, this is also what git does, i.e. "Download X from Y and place it under Z".

Yeah, big Linux distros don't mistreat their devs and ship that file with the git package or the bash-completions package, but that's essentially what you want to do. It boils down to having to do:

1) Support a completions file that's packaged by distros
2) have users go out of their way to install it manually

Considering that we don't have people doing 1) yet, and 2) seems quite unlikely right now, I'd also list this is as a low priority. That being said, if someone jumps into it, I woudln't mind...

**Comment by procrypt** [comment]
Cool, I'd like to work on this later as it's not in our priority list right now :)
